core/cap-0023.md

```diff
@@ -239,146 +293,248 @@ default:
+`CREATE_CLAIMABLE_BALANCE_MALFORMED` if
+
+- `asset` is invalid
+- `amount` is non-positive
+- `threshold` is not between 1 and 255 (inclusive)
+- `weight` on any `condition` is not between 1 and 255 (inclusive)
+
+The behavior of `CreateClaimableBalanceOp` is as follows:
+
+1. Fail with `CREATE_CLAIMABLE_BALANCE_LOW_RESERVE` if the `sourceAccount` does
+   not have at least `min(1, condition.size()-1) * baseReserve` available
+   balance of native asset (see CAP-0003)
+2. Deduct `min(1, condition.size()-1) * baseReserve` of native asset from
+   `sourceAccount`
+3. Fail with `CREATE_CLAIMABLE_BALANCE_NO_TRUST` if the `sourceAccount` does not
+   have a trust line for `asset`
+4. Fail with `CREATE_CLAIMABLE_BALANCE_NOT_AUTHORIZED` if the `sourceAccount` is
+   not authorized for `asset`
+5. Fail with `CREATE_CLAIMABLE_BALANCE_UNDERFUNDED` if the `sourceAccount` does
+   not have at least `amount` available balance of `asset`
+6. Deduct `amount` of `asset` from `sourceAccount`
+7. Create a balance entry with the following properties:
+    - `balanceID` of type `CLAIMABLE_BALANCE_ID_TYPE_V0`. `balanceID.v0()` equal
+       to the SHA256 hash of the concatenation of the `sourceAccount` of the
+       transaction, the `seqNum` of the transaction, and the index of this
+       operation in the transaction
+    - `condition` as specified in the operation, with the exception that
+        - `CLAIM_PREDICATE_BEFORE_RELATIVE_TIME` will be converted to
+          `CLAIM_PREDICATE_BEFORE_ABSOLUTE_TIME` by adding `relBefore` to the
+          `closeTime` in the `LedgerHeader`
+        - `CLAIM_PREDICATE_BEFORE_RELATIVE_TIME` will be converted to
+          `CLAIM_PREDICATE_AFTER_ABSOLUTE_TIME` by adding `relAfter` to the
+          `closeTime` in the `LedgerHeader`
+    - `threshold` as specified in the operation
+    - `reserve` equal to `min(1, condition.size()-1) * baseReserve`
+    - `asset` as specified in the operation
+    - `amount` as specified in the operation
+    - `authorize` equal to 0
+8. Succeed with `CREATE_CLAIMABLE_BALANCE_SUCCESS`
+
+`CreateClaimableBalanceOp` requires medium threshold because it can be used to
+send funds.
+
+### ClaimClaimableBalanceOp
+A `ClaimableBalanceEntry` can only be deleted by the `ClaimClaimableBalanceOp`
+operation. `ClaimClaimableBalanceOp` cannot be malformed. The behavior of
+`ClaimClaimableBalanceOp` is as follows:
+
+1. Fail with `CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST` if there is no
+   `ClaimableBalanceEntry` matching `balanceID`.
+2. Calculate the total weight of `condition` with `destination = sourceAccount`
+   and satisfied predicates.
+3. Fail with `CLAIM_CLAIMABLE_BALANCE_CANNOT_CLAIM` if the total weight is less
+   than `threshold`.
+4. Fail with `CLAIM_CLAIMABLE_BALANCE_LINE_FULL` if the `sourceAccount` does not
+   have at least `reserve` available limit of native asset (see CAP-0003)
+5. Add `reserve` of native asset to the `sourceAccount`
```

Comment:
Because the reserve is always transmitted to the recipient, the sender is losing native asset on every send. The cost of a claimable balance send is significantly more than the cost to send without a claimable. Is that correct? This seems like something that will prohibit its use. Is that intentional to limit the use of this op?